kratos/solving_strategies/schemes/residual_based_bdf2_displacement_scheme.h

```diff
@@ -0,0 +1,600 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+
+#if !defined(KRATOS_RESIDUAL_BASED_BDF2_DISPLACEMENT_SCHEME )
+#define  KRATOS_RESIDUAL_BASED_BDF2_DISPLACEMENT_SCHEME
+
+/* System includes */
+
+/* External includes */
+
+/* Project includes */
+#include "solving_strategies/schemes/residual_implicit_time_scheme.h"
+#include "includes/variables.h"
+#include "includes/checks.h" 
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+///@}
+///@name Type Definitions
+///@{
+///@}
+///@name  Enum's
+///@{
+///@}
+///@name  Functions
+///@{
+///@}
+///@name Kratos Classes
+///@{
+
+/** @brief BDF2 integration scheme (for dynamic problems, displacement based)
+ * @details The second order Backward Differentiation Formula (BDF2) method is a two step second order accurate method. 
+ * This scheme is designed to solve a system of the type:
+ *\f[
+ *   \mathbf{M} \frac{d^2(u_{n0})}{dt^2} + \mathbf{D} \frac{d(un0)}{dt} + \mathbf{K} u_{n0} = \mathbf{f}_{ext}
+ * \f]
+ * 
+ * If we call:
+ * 
+ * - Acelerations: 
+ *      -# \f$ a_{n0} \f$ the acceleration at the current step
+ *      -# \f$ a_{n1} \f$ the acceleration one step in the past
+ *      -# \f$ a_{n2} \f$ the acceleration two steps in the past
+ * - Velocities:
+ *     -# \f$ v_{n0} \f$ the velocity at the current step
+ *     -# \f$ v_{n1} \f$ the velocity one step in the past
+ *     -# \f$ v_{n2} \f$ the velocity two steps in the past
+ * - Displacements:
+ *     -# \f$ u_{n0} \f$ the displacement at the current step
+ *     -# \f$ u_{n1} \f$ the displacement one step in the past
+ *     -# \f$ u_{n2} \f$ the displacement two steps in the past
+ * 
+ * Then we assume:
+ *  \f[ \frac{d(vn0)}{dt} \|t_{n0} = c_0 v_{n0} + c_1 v_{n1} + c_2 v_{n2} \f]
+ *  \f[ \frac{d(un0)}{dt} \|t_{n0} = c_0 u_{n0} + c_1 u_{n1} + c_2 u_{n2} \f]
+ * with:
+ *  -# \f$ c_0 = \frac{1.5}{dt} \f$ 
```

Comment:
please consider the case of variable time steps. take a look e.g. at the processes/compute_bdfcoefficients_process.h
 which has the relations implemented.

Reply:
mmm i think you did implement the variable time step case, however the comment is outdated

Reply:
I consider it later